File: frontend/src/constants/application.ts
export type ApplicationStatus =
  | 'planning'
  | 'preparing'
  | 'submitted'
  | 'waiting'
  | 'admitted'
  | 'rejected'
  | 'waitlisted'

export const ApplicationStatusMap: Record<ApplicationStatus, { text: string; color: string }> = {
  planning: { text: '规划中', color: 'default' },
  preparing: { text: '材料准备中', color: 'blue' },
  submitted: { text: '已提交', color: 'cyan' },
  waiting: { text: '等待结果', color: 'gold' },
  admitted: { text: '已录取', color: 'green' },
  rejected: { text: '已拒绝', color: 'red' },
  waitlisted: { text: '候补名单', color: 'orange' },
}

export const ApplicationStatusList = Object.keys(ApplicationStatusMap) as ApplicationStatus[]

File: backend/internal/constants/application.go
package constants

// ApplicationStatus enumerates the application project state machine.
const (
	AppStatusPlanning  = "planning"
	AppStatusPreparing = "preparing"
	AppStatusSubmitted = "submitted"
	AppStatusWaiting   = "waiting"
	AppStatusAdmitted  = "admitted"
	AppStatusRejected  = "rejected"
	AppStatusWaitlisted = "waitlisted"
)

// ValidApplicationStatuses returns all accepted statuses.
func ValidApplicationStatuses() []string {
	return []string{
		AppStatusPlanning, AppStatusPreparing, AppStatusSubmitted,
		AppStatusWaiting, AppStatusAdmitted, AppStatusRejected, AppStatusWaitlisted,
	}
}

// IsValidApplicationStatus reports whether a status is known.
func IsValidApplicationStatus(s string) bool {
	for _, v := range ValidApplicationStatuses() {
		if v == s {
			return true
		}
	}
	return false
}

// NextApplicationStatuses returns allowed forward transitions.
func NextApplicationStatuses(s string) []string {
	switch s {
	case AppStatusPlanning:
		return []string{AppStatusPreparing}
	case AppStatusPreparing:
		return []string{AppStatusSubmitted}
	case AppStatusSubmitted:
		return []string{AppStatusWaiting}
	case AppStatusWaiting:
		return []string{AppStatusAdmitted, AppStatusRejected, AppStatusWaitlisted}
	default:
		return nil
	}
}
